modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/Archetyper.java

```diff
@@ -78,224 +46,68 @@
 	}
 
-	private static File _getArchetypeFile(String artifactId, File file)
-		throws IOException {
-
-		try (JarFile jarFile = new JarFile(file)) {
-			Enumeration<JarEntry> enumeration = jarFile.entries();
-
-			while (enumeration.hasMoreElements()) {
-				JarEntry jarEntry = enumeration.nextElement();
-
-				if (jarEntry.isDirectory()) {
-					continue;
-				}
-
-				String name = jarEntry.getName();
-
-				if (!name.startsWith(artifactId + "-")) {
-					continue;
-				}
-
-				Path archetypePath = Files.createTempFile(
-					"temp-archetype", null);
-
-				Files.copy(
-					jarFile.getInputStream(jarEntry), archetypePath,
-					StandardCopyOption.REPLACE_EXISTING);
-
-				File archetypeFile = archetypePath.toFile();
-
-				archetypeFile.deleteOnExit();
-
-				return archetypeFile;
-			}
-		}
-
-		return null;
+	public Archetyper(List<File> archetypesDirs) {
+		_archetypesDirs = archetypesDirs;
 	}
 
-	private static File _getTemplateFile(
-			ProjectTemplatesArgs projectTemplatesArgs)
+	public ArchetypeArtifactManager createArchetypeArtifactManager()
 		throws Exception {
 
-		String template = projectTemplatesArgs.getTemplate();
-
-		for (File archetypesDir : projectTemplatesArgs.getArchetypesDirs()) {
-			if (!archetypesDir.isDirectory()) {
-				continue;
-			}
-
-			try (DirectoryStream<Path> directoryStream =
-					Files.newDirectoryStream(
-						archetypesDir.toPath(), "*.project.templates.*")) {
-
-				for (Path path : directoryStream) {
-					String fileName = String.valueOf(path.getFileName());
-
-					String templateName = ProjectTemplatesUtil.getTemplateName(
-						fileName);
-
-					if (templateName.equals(template)) {
-						return path.toFile();
-					}
-				}
-			}
-		}
-
-		File archetypesFile = FileUtil.getJarFile(ProjectTemplates.class);
+		ArchetypeArtifactManager archetypeArtifactManager =
+			newArchetypeArtifactManager();
 
-		String artifactId =
-			ProjectTemplates.TEMPLATE_BUNDLE_PREFIX +
-				template.replace('-', '.');
+		ReflectionUtil.setFieldValue(
+			_loggerField, archetypeArtifactManager, _logger);
 
-		return _getArchetypeFile(artifactId, archetypesFile);
+		return archetypeArtifactManager;
 	}
 
-	private static boolean _isInVersionRange(
-		String versionString, String range) {
-
-		Version version = new Version(versionString);
+	public ArchetypeManager createArchetypeManager() throws Exception {
+		DefaultArchetypeManager archetypeManager =
+			new DefaultArchetypeManager();
 
-		VersionRange versionRange = new VersionRange(range);
+		ReflectionUtil.setFieldValue(_loggerField, archetypeManager, _logger);
+		ReflectionUtil.setFieldValue(
+			DefaultArchetypeManager.class, "generator", archetypeManager,
+			_createArchetypeGenerator());
 
-		return versionRange.includes(version);
+		return archetypeManager;
 	}
 
-	private ArchetypeArtifactManager _createArchetypeArtifactManager(
-			List<File> archetypesDirs)
+	protected ArchetypeArtifactManager newArchetypeArtifactManager()
```

Comment:
Only use `protected` if needed, otherwise it should be `private`.

Please rename the method to `createArchetypeArtifactManager` for consistency with the other methods in this class.

Reply:
We need it to be protected because a new sub-class called `CodeGenerator` that we are still building will need to override the creation of the instance class.

Also we can't rename it to `createArchetypeArtrifactManager` because there already is one method with that name.